kura/test/org.eclipse.kura.nm.test/src/test/java/org/eclipse/kura/nm/configuration/NMSettingsConverterTest.java

```diff
@@ -1382,4 +1709,27 @@ public Object buildAddressDataWith(String ipAddr, UInt32 prefix) {
 
         return dataVariant.getValue();
     }
+
+    public List<List<Byte>> buildDnsAddressesList(String inputDns) {
+        List<List<Byte>> dnsAddresses = new ArrayList<>();
+
+        InetAddress address;
+
+        try {
+            address = InetAddress.getByName(inputDns);
+        } catch (UnknownHostException ex) {
+            address = null;
+        }
+
+        byte[] dnsByteArray = address.getAddress();
+
+        List<Byte> dnsAddress = new ArrayList<>();
+        for (int i = 0; i < dnsByteArray.length; i++) {
+            dnsAddress.add(dnsByteArray[i]);
+        }
+
+        dnsAddresses.add(dnsAddress);
+
+        return dnsAddresses;
+    }
```

Comment:
This is wrong on a fundamental level.

You cannot test your code with the code you're testing. See [our docs](https://eurotech.atlassian.net/wiki/spaces/ECE/pages/2271510531/Mattia+and+Greg+s+Unit+Testing+Tips#Guideline-2:-Prefer-static-data-in-expected-values)

> If we replicate the code logic in our tests we would need unit tests for our unit tests. The unit tests should be our single source of truth about the expected output given an input.

Substitute this method call with a constant wherever you used this method and get rid of this method.

_Note_: Yes, I noticed that there's a complementary method `buildAddressDataWith` that performs the same. That method is wrong too. We should weed that out. Ignore it for now, we'll take care of that in a future PR.